engine/gamesys/src/gamesys/scripts/script_tilemap.cpp

```diff
@@ -263,10 +264,25 @@ namespace dmGameSystem
             assert(top + 1 == lua_gettop(L));
             return 1;
         }
-
-        bool flip_h = lua_toboolean(L, 6);
-        bool flip_v = lua_toboolean(L, 7);
-        SetTileGridTile(component, layer_index, cell_x, cell_y, tile, flip_h, flip_v);
+        bool flip_h = 0;
+        bool flip_v = 0;
+        bool rotated90 = 0;
+        if (lua_isnumber(L, 6))
+        {
+            int bitmask = dmMath::Abs(luaL_checkinteger(L, 6));
+            flip_h = (bitmask & 1) == 1;
+            flip_v = (bitmask & 2) == 2;
+            rotated90 = (bitmask & 4) == 4;
```

Comment:
Given the 6 different bit as input, I would have thought we would have a more complicated setup here to convert it back to 3 bits.

I'm assuming it has to do with the -3 and -7 values below?

Although it might work, I think it's just slightly too complicated to read and reason about, and I'd just rather have the logic "spelled out"

Reply:
If using the R180 = H_FLIP + V_FLIP approach, this part needs could be left as-is, if there is also a comment explaining the bit logic here.

Reply:
I added more comments when create constants